app/validation/validators.py

```diff
@@ -68,21 +68,28 @@ class NumberRange(object):
         The maximum value of the number. If not provided, maximum value
         will not be checked.
     """
-    def __init__(self, minimum=None, maximum=None, messages=None):
+    def __init__(self, minimum=None, maximum=None, messages=None, format_type=None):
         self.minimum = minimum
         self.maximum = maximum
         if not messages:
             messages = error_messages
         self.messages = messages
+        self.format_type = format_type
 
     def __call__(self, form, field):
         data = field.data
         error_message = None
         if data is not None:
             if self.minimum is not None and data < self.minimum:
-                error_message = self.messages['NUMBER_TOO_SMALL'] % dict(min=format_number(self.minimum))
+                if self.format_type == 'Currency':
```

Comment:
it might be nicer to modify `format_number` to take in `format_type` and then do the formatting stitch to currency in their.
It would simplify this if conditions here a lot.